contracts/modules/TransferManager/VolumeDumpingRestrictionManager.sol

```diff
@@ -0,0 +1,322 @@
+    * @param _endTime When the dumping restriction ends in seconds
+    * @param _rollingPeriod Ttime period in seconds
+    */
+    function addDumpingRestriction(
+        address _userAddress, 
+        uint256 _percentAllowed, 
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriod ) public withPerm(ADMIN) {
+
+        _checkVolumeDumpingParams(_userAddress, _percentAllowed, _startTime, _endTime, _rollingPeriod);
+        
+        // deny if restriction already exists
+        require(volumeRestriction[_userAddress].percentAllowed == 0, "volume dumping restriction already exists");
+        
+        // if a startTime of 0 is passed in, then start now.
+        if(_startTime == 0) {
+            _startTime = now;
+            require(_endTime > _startTime, "End time must be greater than start time");
+        }
+        
+        // add the volume dumping restriction
+        volumeRestriction[_userAddress] = VolumeRestriction(_percentAllowed, _startTime, _endTime, _rollingPeriod);
+
+        emit AddNewVolumeDumping(
+            _userAddress,
+            _percentAllowed,
+            _startTime,
+            _endTime,
+            _rollingPeriod
+        );
+    }
+
+    /**
+    * @notice lets the admin create multiple volume dumping restriction for a multiple addresses.
+    * @param _userAddresses Address of the user to apply the volume dumping restriction
+    * @param _percentsAllowed Percent of tokens balance allowed to transfer within a rolling period
+    * @param _startTimes When the dumping restriction, 0 means now in seconds
+    * @param _endTimes When the dumping restriction ends in seconds
+    * @param _rollingPeriods Time period in seconds
+    */
+    function addDumpingRestrictionMulti(
+        address[] _userAddresses,
+        uint256[] _percentsAllowed, 
+        uint256[] _startTimes, 
+        uint256[] _endTimes, 
+        uint256[] _rollingPeriods) public withPerm(ADMIN) {
+        require( 
+            _userAddresses.length == _percentsAllowed.length &&
+            _userAddresses.length == _rollingPeriods.length &&
+            _userAddresses.length == _startTimes.length &&
+            _userAddresses.length == _endTimes.length,
+            "Input array length mis-match"
+        );
+
+        for (uint i = 0; i < _userAddresses.length; i++) {
+            addDumpingRestriction(_userAddresses[i], _percentsAllowed[i], _startTimes[i], _endTimes[i], _rollingPeriods[i]);
+        }
+    }
+
+     /**
+    * @notice lets the admin remove a volume dumping restriction.
+    * @param _userAddress User address to remove the volume dumping restriction
+    */
+    function removeRestriction(address _userAddress) public withPerm(ADMIN) {
+        VolumeRestriction memory toRemove = volumeRestriction[_userAddress];
+
+        // check if entry is in the past
+        // not allowed to delete past entries
+        require(toRemove.endTime > now, "Cannot remove past restrictions");
```

Comment:
wrong reason string. It should be `Only past restrictions are allowed to delete` more preferable is `Only passed restrictions are allowed to delete`

Reply:
According to the flow chart design in the[Lucid Chart](https://www.lucidchart.com/documents/edit/c96ff64c-5b89-4b9c-985a-01637266bf88/0?shared=true&) It says below
![image](https://user-images.githubusercontent.com/8148384/46685907-893e0c00-cbee-11e8-9b4a-7ebd0905da1c.png)